charts/du-metrics-server/values.yaml

```diff
@@ -0,0 +1,94 @@
+du-metrics-server:
+  config:
+    testbed: default
+
+  replicaCount: 1
+
+  initImage:
+    influxdb:
+      repository: accelleran/acc-generic-img
+      pullPolicy: IfNotPresent
+      tag: "0.8.0"
+
+  image:
+    repository: accelleran/du-metrics-server
+    pullPolicy: IfNotPresent
+    # Overrides the image tag whose default is the chart appVersion.
+    tag: ""
+
+  imagePullSecrets:
+    - name: accelleran-secret
+
+  accelleranLicense:
+    enabled: false
+
+  extraEnvs: []
+
+  extraLabels: {}
+  annotations: {}
+
+  podSecurityContext:
+    runAsNonRoot: true
+    runAsUser: 1000
+    fsGroup: 1000
+
+  securityContext: {}
+    # capabilities:
+    #   drop:
+    #   - ALL
+    # readOnlyRootFilesystem: true
+    # runAsNonRoot: true
+    # runAsUser: 1000
+
+  resources: {}
+    # limits:
+    #   cpu: 100m
+    #   memory: 128Mi
+    # requests:
+    #   cpu: 100m
+    #   memory: 128Mi
+
+  nodeSelector: {}
+
+  tolerations: []
+
+  affinity: {}
+
+  service:
+    enabled: true
+    name: ""
+    type: NodePort
+    ports:
+      metric:
+        port: 55555
+        targetPort: 55555
+        nodePort: 31555
+        protocol: UDP
+
+
+influxdb:
+  enabled: true
+
+  adminUser:
+    organization: "accelleran"
+    bucket: "default"
+    retention_policy: "0s"
+    user: "admin"
+    password: "password"
+    token: "ubZW5JVsnhjaLTMeBSYVjvTc4pkJ5jPYk8Tqkj2R9j28Ern4kaZP4yPo4bHbLfYG"
```

Comment:
Shouldn't we generate these? (And perhaps then store them in a secret?)

Reply:
It's the default behaviour that these are generated, but the problem is that in some way it's also stored or something as an upgrade broke access to influxdb. Only way to recover was to remove the pvc and reinstall. I'll test maybe once more with that in the current version of these charts, but otherwise we could also opt to let the user provide a secret (`existingSecret`) to keep the password and token stable

Reply:
Now it's stable with upgrades, so we use a randomly generated value then. We'll need to clearly document then that a `helm uninstall` needs to go together with removing the pvc. This was probably the state then in which I saw this happening and I was just mixing the two up...

Reply:
[This is how Bitnami handles it](https://docs.bitnami.com/general/how-to/troubleshoot-helm-chart-issues/#credential-errors-while-upgrading-chart-releases), at least while upgrading the release:

> ```
> export POSTGRESQL_PASSWORD=$(kubectl get secret --namespace default MY-RELEASE-postgresql -o jsonpath="{.data.postgresql-password}" | base64 --decode)
> helm upgrade MY-RELEASE oci://registry-1.docker.io/bitnamicharts/postgresql --set postgresqlPassword=$POSTGRESQL_PASSWORD
> ```

Reply:
It's already handled completely by influxdb. They use the `lookup` function to get the data of the secret from previous deployments: https://helm.sh/docs/chart_template_guide/function_list/#lookup

Reply:
Fancy 😉